### PR TITLE
VR-6131 Add self_contained param to download_docker_context()

### PR DESCRIPTION
**Pull request by nhatsmrt**

### CODE REVIEW

File: client/verta/verta/_registry/modelversion.py [hunk 1]
```diff
@@ -670,14 +670,16 @@ def get_labels(self):
...
-    def download_docker_context(self, download_to_path):
+    def download_docker_context(self, download_to_path, self_contained=False):
         """
         Downloads this Model Version's Docker context ``tgz``.
 
         Parameters
         ----------
         download_to_path : str
             Path to download Docker context to.
+        self_contained : bool
```
Comment (convoliution) [inline client/verta/verta/_registry/modelversion.py]:
```suggestion
        self_contained : bool, default False
```

File: client/verta/verta/_registry/modelversion.py [hunk 2]
```diff
@@ -691,7 +693,8 @@ def download_docker_context(self, download_to_path):
             self._conn.socket,
         )
         body = {
-            "model_version_id": self.id
+            "model_version_id": self.id,
+            "self_contained": self_contained
```
Comment (convoliution) [inline client/verta/verta/_registry/modelversion.py]: Personal preference: trailing commas makes future diffs easier on the eyes
```suggestion
            "self_contained": self_contained,
```

File: client/verta/verta/_tracking/experimentrun.py [hunk 3]
```diff
@@ -2101,27 +2101,34 @@ def download_deployment_yaml(self, download_to_path, path=None, token=None, no_t
...
-    def download_docker_context(self, download_to_path):
+    def download_docker_context(self, download_to_path, self_contained=False):
         """
         Downloads this Experiment Run's Docker context ``tgz``.
 
         Parameters
         ----------
         download_to_path : str
             Path to download Docker context to.
+        self_contained : bool
```
Comment (convoliution) [inline client/verta/verta/_tracking/experimentrun.py]:
```suggestion
        self_contained : bool, default False
```

File: client/verta/verta/_tracking/experimentrun.py [hunk 4]
```diff
@@ -2101,27 +2101,34 @@ def download_deployment_yaml(self, download_to_path, path=None, token=None, no_t
...
+        self._refresh_cache()
+        endpoint = "{}://{}/api/v1/deployment/builds/dockercontext".format(
             self._conn.scheme,
             self._conn.socket,
-            self.id,
         )
-        with _utils.make_request("GET", endpoint, self._conn, stream=True) as response:
+        body = {
+            "run_id": self.id,
+            "self_contained": self_contained
```
Comment (convoliution) [inline client/verta/verta/_tracking/experimentrun.py]:
```suggestion
            "self_contained": self_contained,
```

**Review by convoliution — APPROVED** [review]
@nhatsmrt Did you happen to happen to also find and run the existing basic tests for these?

File: client/verta/verta/_registry/modelversion.py (hunk 1 above)
Comment (nhatsmrt) [reply client/verta/verta/_registry/modelversion.py]: Fixed.

File: client/verta/verta/_registry/modelversion.py (hunk 2 above)
Comment (nhatsmrt) [reply client/verta/verta/_registry/modelversion.py]: Fixed.

File: client/verta/verta/_tracking/experimentrun.py (hunk 3 above)
Comment (nhatsmrt) [reply client/verta/verta/_tracking/experimentrun.py]: Fixed.

File: client/verta/verta/_tracking/experimentrun.py (hunk 4 above)
Comment (nhatsmrt) [reply client/verta/verta/_tracking/experimentrun.py]: Fixed.

**Comment by nhatsmrt** [comment]
I did run the download docker context tests for Model Version and Experiment Run. Also ran a download test in test_cli/test_endpoint. All passed. It should work for both self_contained = True and False (since I originally used True as default, then switched to False and re-ran the tests).

**Review by convoliution — APPROVED** [review]
Works for me!

**Pull request merged**